core/node/external_proof_integration_api/src/metrics.rs

```diff
@@ -0,0 +1,60 @@
+use std::time::Duration;
+
+use tokio::time::Instant;
+use vise::{Counter, EncodeLabelSet, EncodeLabelValue, Family, Histogram, LabeledFamily, Metrics};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "outcome", rename_all = "snake_case")]
+pub(crate) enum CallOutcome {
+    Success,
+    Failure,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "type", rename_all = "snake_case")]
+pub(crate) enum MethodType {
+    GetLatestProofGenerationData,
+    GetSpecificProofGenerationData,
+    VerifyProof,
+}
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "external_proof_integration_api")]
+pub(crate) struct ProofIntegrationApiMetrics {
+    pub call_count: Family<MethodType, Counter<u64>>,
+    #[metrics(labels = ["type", "outcome"])]
+    pub call_outcome: LabeledFamily<(MethodType, CallOutcome), Counter<u64>, 2>,
+    #[metrics(buckets = vise::Buckets::LATENCIES)]
+    pub call_latency: Family<MethodType, Histogram<Duration>>,
```

Comment:
with this you don't really need the two metrics above. Histogram always creates a `counter` as well.